fv/wireguard_test.go

```diff
@@ -875,9 +880,260 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported 3 node
+						fmt.Sprintf("%s dev %s scope link", felixes[j].IP, wireguardInterfaceNameDefault)))
+					// check for routes to pods on other felix nodes
+					for _, wl := range wls {
+						matchers = append(matchers, ContainSubstring(
+							fmt.Sprintf("%s dev %s scope link", wl.IP, wireguardInterfaceNameDefault)))
+					}
+				}
+			}
+			Eventually(func() []string {
+				return strings.Split(getWireguardRouteEntry(felixes[i]), "\n")
+			}, "10s", "100ms").Should(ContainElements(matchers))
+		}
+
+		By("Checking the proc/sys src valid mark entries")
+		for _, felix := range felixes {
+			Eventually(func() string {
+				s, _ := felix.ExecCombinedOutput("cat", "/proc/sys/net/ipv4/conf/all/src_valid_mark")
+				return s
+			}, "10s", "100ms").Should(ContainSubstring("1"))
+		}
+
+		By("Checking wireguard allowed ips")
+		for i, _ := range wlsByHost {
+			var matchers []types.GomegaMatcher
+			for j, wls := range wlsByHost {
+				if i != j {
+					var allowedIPMatchers []types.GomegaMatcher
+					// check for allowed IP entry for other felix nodes
+					allowedIPMatchers = append(allowedIPMatchers, ContainSubstring(felixes[j].IP))
+					// check for routes to pods on other felix nodes
+					for _, wl := range wls {
+						allowedIPMatchers = append(allowedIPMatchers, ContainSubstring(wl.IP))
+					}
+					matchers = append(matchers, And(allowedIPMatchers...))
+				}
+			}
+			Eventually(func() []string {
+				s, _ := felixes[i].ExecCombinedOutput("wg", "show", wireguardInterfaceNameDefault, "allowed-ips")
+				return strings.Split(s, "\n")
+			}, "10s", "100ms").Should(ContainElements(matchers))
+		}
+
+		//By("Ensuring felix nodes can connect to each other")
+		//for i := range felixes {
+		//	for j := range felixes {
+		//		if i != j {
+		//			Eventually(func() bool {
+		//				return felixes[i].CanConnectTo(felixes[j].IP, defaultWorkloadPort, "tcp").HasConnectivity()
+		//			}, "10s", "100ms").Should(BeTrue())
+		//		}
+		//	}
+		//}
+
+		By("verifying packets between felix-0 and felix-1 is encrypted")
+		cc.ExpectSome(wlsByHost[0][0], wlsByHost[1][0])
+		cc.ExpectSome(wlsByHost[1][0], wlsByHost[0][0])
+		cc.CheckConnectivityWithTimeout(30 * time.Second)
+		for i := range []int{0, 1} {
+			Eventually(tcpdumps[i].MatchCountFn("numTunnelPacketsFelix0toFelix1"), "10s", "100ms").
+				Should(BeNumerically(">", 0))
+			Eventually(tcpdumps[i].MatchCountFn("numTunnelPacketsFelix1toFelix0"), "10s", "100ms").
+				Should(BeNumerically(">", 0))
+
+			Eventually(tcpdumps[i].MatchCountFn("numNonTunnelPacketsFelix0toFelix1"), "10s", "100ms").
+				Should(BeNumerically("==", 0))
+			Eventually(tcpdumps[i].MatchCountFn("numNonTunnelPacketsFelix1toFelix0"), "10s", "100ms").
+				Should(BeNumerically("==", 0))
+		}
+
+		By("checking same node pod-to-pod connectivity")
+		cc.ResetExpectations()
+		for felixIdx := 0; felixIdx < nodeCount; felixIdx++ {
+			cc.ExpectSome(wlsByHost[felixIdx][0], wlsByHost[felixIdx][1])
+		}
+		cc.CheckConnectivity()
```

Comment:
Any reason to do each step of these checks separately?  The connectivity checker is intended to do several checks in parallel for performance so we usually just add all the expectations and then call CheckConnectivity() once/

Reply:
No, I've now combined them as you suggested.